loki/transform/transform_array_indexing.py

```diff
@@ -33,9 +33,16 @@
 ]
 
 
-def shift_to_zero_indexing(routine):
+def shift_to_zero_indexing(routine, ignore=()):
```

Comment:
It's generally preferred to assign `None` as default value for optional arguments, and then convert it to a tuple in the function body, e.g. `ignore = as_tuple(ignore)`.
That also adds the benefit that you can do something like `shift_to_zero_indexing(routine, ignore='myvar')` and wouldn't get the unexpected behaviour that this is treated as `ignore=('m', 'y', 'v', 'a', 'r')`.